src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -525,6 +525,135 @@ public override BoundNode VisitTypeOrInstanceInitializers(BoundTypeOrInstanceIni
             return new BoundStatementList(node.Syntax, rewrittenStatements, node.HasErrors);
         }
 
+        public override BoundNode VisitArrayAccess(BoundArrayAccess node)
+        {
+            // If the array access index is of type System.Index or System.Range
+            // we need to translate it into a call into an actual indexer, instead
+            // of a simple array element access.
+
+            if (this.EmitModule is null || 
```

Comment:
Why do we need to check for EmitModule being null?

Reply:
We shouldn't have to any more. That was a remnant of an earlier approach.